daft/io/writer.py

```diff
@@ -1,21 +1,31 @@
 import uuid
 from abc import ABC, abstractmethod
-from typing import Optional
+from typing import TYPE_CHECKING, List, Optional, Tuple
 
 from daft.daft import IOConfig
-from daft.dependencies import pa, pacsv, pq
+from daft.dependencies import pa, pacsv, pafs, pq
 from daft.filesystem import (
     _resolve_paths_and_filesystem,
     canonicalize_protocol,
     get_protocol_from_path,
 )
+from daft.iceberg.iceberg_write import (
+    coerce_pyarrow_table_to_schema,
+    make_iceberg_data_file,
+    make_iceberg_record,
+)
 from daft.series import Series
 from daft.table.micropartition import MicroPartition
 from daft.table.partitioning import (
     partition_strings_to_path,
     partition_values_to_str_mapping,
 )
 from daft.table.table import Table
+from daft.table.table_io import make_deltalake_add_action, make_deltalake_fs, sanitize_table_for_deltalake
```

Comment:
Let's move these functions out into their own file like they are done for Iceberg